### PR TITLE
Tobin riku pipe stats feature check

### LINKED ISSUE #1179: Pipeline statistics query feature is not checked by validation
Validation layers fail to notify of problems when creating and using pipeline statistics queries without enabling the "pipelineStatisticsQuery" feature when creating device.

The validator should check when vkCreateQueryPool is called with VK_QUERY_TYPE_PIPELINE_STATISTICS that .pEnabledFeatures.pipelineStatisticsQuery is enabled when calling vkCreateDevice.

### PR DESCRIPTION
**Pull request by tobine**
Fixes #1179 

New check from Riku Salminen to make sure that pipeline stats are enabled when creating a query pool for them.

I added a test to verify that the check works.

### CODE REVIEW

**Comment by karl-lunarg** [comment]
Database update?

**Comment by tobine** [comment]
@karl-lunarg yep, just pushed CL with database update

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -9058,6 +9058,56 @@ TEST_F(VkLayerTest, InvalidDSUpdateIndex) {
...
+    const std::vector<VkQueueFamilyProperties> queue_props = m_device->queue_props;
+    std::vector<VkDeviceQueueCreateInfo> queue_info;
+    queue_info.reserve(queue_props.size());
+    std::vector<std::vector<float>> queue_priorities;
+    for (uint32_t i = 0; i < (uint32_t)queue_props.size(); i++) {
+        VkDeviceQueueCreateInfo qi{};
+        qi.sType = VK_STRUCTURE_TYPE_DEVICE_QUEUE_CREATE_INFO;
+        qi.queueFamilyIndex = i;
+        qi.queueCount = queue_props[i].queueCount;
+        queue_priorities.emplace_back(qi.queueCount, 0.0f);
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: This noise is unfortunate...

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -9058,6 +9058,56 @@ TEST_F(VkLayerTest, InvalidDSUpdateIndex) {
     vkDestroyDescriptorPool(m_device->device(), ds_pool, NULL);
 }
 
+TEST_F(VkLayerTest, InvalidQueryPoolCreate) {
+    TEST_DESCRIPTION("Attempt to create a query pool for PIPELINE_STATISTICS without enabling pipeline stats for the device.");
+
+    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, VALIDATION_ERROR_01006);
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: Just before the call that's expected to produce the error would be clearer.

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**